### PR TITLE
Badly written test corrected

### PR DESCRIPTION
**Pull request by Kwestor**
Disclaimer required by the lawyers:

THE FOLLOWING DISCLAIMER APPLIES TO ALL SOFTWARE CODE AND OTHER MATERIALS CONTRIBUTED IN CONNECTION WITH THIS SOFTWARE:
THIS SOFTWARE IS LICENSED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS “AS IS” AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE AND ANY WARRANTY OF NON-INFRINGEMENT, ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. THIS SOFTWARE MAY BE REDISTRIBUTED TO OTHERS ONLY BY EFFECTIVELY USING THIS OR ANOTHER EQUIVALENT DISCLAIMER IN ADDITION TO ANY OTHER REQUIRED LICENSE TERMS.
ONLY THE SOFTWARE CODE AND OTHER MATERIALS CONTRIBUTED IN CONNECTION WITH THIS SOFTWARE, IF ANY, THAT ARE ATTACHED TO (OR OTHERWISE ACCOMPANY) THIS SUBMISSION (AND ORDINARY COURSE CONTRIBUTIONS OF FUTURES PATCHES THERETO) ARE TO BE CONSIDERED A CONTRIBUTION. NO OTHER SOFTWARE CODE OR MATERIALS ARE A CONTRIBUTION.

### CODE REVIEW

File: org.scala-ide.sdt.debug.expression.tests/src/org/scalaide/debug/internal/expression/features/PartialFunctionLambdaTest.scala [hunk 1]
```diff
@@ -20,11 +20,9 @@ class PartialFunctionLambdaTest extends BaseIntegrationTest(PartialFunctionLambd
   def testPartialFunctionAndAnnotatedPrimitives(): Unit =
     eval("list.filter { case a: Int => a > 2 }", List(3), Scala.::)
 
-  @Ignore("TODO - O-5266 - add support for multiple parameter typed partial function lambda")
   @Test
   def testPartialFunctionAndMulipleParameterLists(): Unit =
-    eval("list.foldLeft(1){case (a: Int, b: Int) => a + b}", 7, Java.boxed.Integer)
-
+    eval("list.zip(list).map { case (a: Int, b: Int) => a + b }", List(2, 4, 6), Scala.::)
```
Comment (sschaef) [inline org.scala-ide.sdt.debug.expression.tests/src/org/scalaide/debug/internal/expression/features/PartialFunctionLambdaTest.scala]: Actually, that is not a PartialFunction.

Comment (sschaef) [inline org.scala-ide.sdt.debug.expression.tests/src/org/scalaide/debug/internal/expression/features/PartialFunctionLambdaTest.scala]: I had a look to the other tests, they are also not PartialFunctions, but Functions that use pattern matching. You could rename the test.

Comment (Kwestor) [inline org.scala-ide.sdt.debug.expression.tests/src/org/scalaide/debug/internal/expression/features/PartialFunctionLambdaTest.scala]: From what I know about Scala, `{ case a: Int => a > 2 }` is a `PartialFunction` literal. It is complete (as in "all matches arguments" from list), but that is determined at runtime.

Comment (Kwestor) [inline org.scala-ide.sdt.debug.expression.tests/src/org/scalaide/debug/internal/expression/features/PartialFunctionLambdaTest.scala]: And this test before re-write failed typecheck with:
```
type mismatch;
 found   : PartialFunction[Any,Int]
 required: (Int, Int) => Int
```

**Comment by dragos** [comment]
LGTM, though I don't see why the previous version was "bad" :)

**Comment by Kwestor** [comment]
I thought previous version should not compile in standard Scala, but I checked now and it does. I don't know how still ;)

**Comment by dragos** [comment]
The answer is pretty tricky to find, but it's auto-tupling. See [Jason's answer](http://stackoverflow.com/questions/2850902/scala-coalesces-multiple-function-call-parameters-into-a-tuple-can-this-be-di).

File: org.scala-ide.sdt.debug.expression.tests/src/org/scalaide/debug/internal/expression/features/PartialFunctionLambdaTest.scala (hunk 1 above)
Comment (sschaef) [inline org.scala-ide.sdt.debug.expression.tests/src/org/scalaide/debug/internal/expression/features/PartialFunctionLambdaTest.scala]: The spec calls it a [pattern matching anonymous function](http://www.scala-lang.org/files/archive/spec/2.11/08-pattern-matching.html#pattern-matching-anonymous-functions). I don't know though why the error message shown something about PartialFunction.

**Pull request merged**